react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -1,88 +1,57 @@
 
 import React, {Component} from 'react';
+import { connect } from 'react-redux';
 
 import styles from './styles.module.scss';
 import Board from './components/Board';
-import {LINES} from './constants';
+import MatchHistory from './components/MatchHistory';
+import { calculateWinner } from "~utils";
+import gameActionsCreator from "~redux/game/actions";
+import { JUMP_TEXT, GAME_START_TEXT, NEXT_PLAYER, WINNER, TIC, TAC } from "~constants";
 
 class Game extends Component {
-  state = {
-    history: [{ squares: Array(9).fill(null)}],
-    stepNumber: 0,
-    xIsNext: true
-  }
-
-  clickHandler = i => {
-    const { history, stepNumber, xIsNext} = this.state;
-    const auxHistory = history.slice(0, stepNumber + 1);
-    const current = auxHistory[auxHistory.length -1];
-    const squares = current.squares.slice();
-    if (this.calculateWinner(squares) || squares[i]) {
-      return;
-    }
-    squares[i] = xIsNext ? 'X' : 'O';
-    this.setState( prevState => ({ 
-      history: auxHistory.concat([{squares}]),
-      stepNumber: auxHistory.length,
-      xIsNext: !prevState.xIsNext 
-      })
-    );
-  }
-
-  jumpTo = step => this.setState({
-      stepNumber: step,
-      xIsNext: (step % 2) === 0
-    });
-
-  calculateWinner = squares => {
-    for (let i = 0; i < LINES.length; i++) {
-      const [a, b, c] = LINES[i];
-      if (squares[a] && 
-      squares[a] === squares[b] && 
-      squares[a] === squares[c]) {
-        return squares[a];
-      }
-    }
-    return null;
-  }
 
   render(){
-    const history = this.state.history;
-    const current = history[this.state.stepNumber];
-    const winner = this.calculateWinner(current.squares);
-
+    const { current, jumpTo, xIsNext } = this.props;
+    const history = this.props.history;
+    const winner = calculateWinner(current.squares);
     const moves = history.map((step,move) => {
       const desc = move ?
-        'Go to move #' + move :
-        'Go to game start';
+        JUMP_TEXT + move :
```

Comment:
You can change JUMP_TEXT to a function that receive the move and return the string